### DISCUSSION TITLE
VS Code Multiline Execution

### DISCUSSION THREAD

**Discussion by EKtheSage** [quarto-dev/quarto-cli#3679] [Q&A]
Currently there are a few ways to execute commands inside VS Code Quarto extension:

Both `SHIFT + ENTER` and `CTRL + ENTER` execute the current line the cursor is on, or the highlighted multiple lines of code.
`CTRL + SHIFT + ENTER` executes the entire cell.

In RStudio, there is an option to execute multi-line statement without highlighting first:

![image](https://user-images.githubusercontent.com/7637535/207830215-fbacd991-db51-416d-9bf9-628160bddb53.png)

Do we have similar functionality inside VS Code? This will make executing code much more convenient since we can skip highlighting the entire multiline code statement before running `CTRL+ENTER`

The other approach is to have one statement per code cell, and use `CTRL + SHIFT + ENTER` to run a multiline code block, but that is still more inconvenient that RStudio's feature.

**Comment by mcanouil** [comment]
Yes, see <https://github.com/quarto-dev/quarto-vscode/issues/108>.
And as usual with VSCode check/set your keybindings commands (`quarto.runSelection` as a filter to find if any already set, which should be the case since defined as default in the extension).

**Reply by EKtheSage** [reply]
This works for R, but does it work for Python? I have version Quarto 1.57 but I still cannot run a block of Python code.

**Reply by mcanouil** [reply]
Then, it might not be the case as for Julia, but you already know that since you kind of cross-posted this on <https://github.com/quarto-dev/quarto-vscode/issues/7>.

Quarto CLI is agnostic to the language, meaning any languages can be used.
Quarto VSCode extension on the other hand in an ideal world, would allow the same feature for every languages supported in Quarto, but you can easily see that this is not at all possible.
Quarto VSCode extension is basically borrowing stuff from other extension.

If a Python or Jupyter extension can do the multi-line thing, then you can activate the extension for Quarto document either as a one time thing or permanently.
Otherwise, I am afraid you're gonna have to wait.

**Comment by kvten** [comment]
Here are my keyboard shortcuts:

![Screenshot](https://user-images.githubusercontent.com/103538771/209727995-a61926dd-ed53-4678-96e3-192eff703ea1.png)

Although they're listed under quarto, you'll have to change them through VS Code.

`Preferences` > `Keyboard shortcuts` > double click on command > enter keystrokes > save
